clang/include/clang/Basic/Attr.td

```diff
@@ -279,12 +279,13 @@ class DefaultIntArgument<string name, int default> : IntArgument<name, 1> {
 
 // This argument is more complex, it includes the enumerator type name,
 // a list of strings to accept, and a list of enumerators to map them to.
-class EnumArgument<string name, string type, list<string> values,
+class EnumArgument<string name, string type, bit ext, list<string> values,
```

Comment:
```suggestion
class EnumArgument<string name, string type, bit external_type, list<string> values,
```

Also, you need to update the comment on `EnumArgument`.  The name suggestion is an attempt to make it more clear what this means.

Also: I see we have values at the 'end' with default, instead of doing all the mods in `Attr.td`, can we move this external to the end as well with a default 'false' value?

Reply:
Should we also consider updating https://clang.llvm.org/docs/InternalsManual.html#arguments to explain this particular argument in more detail given that it's a pretty complicated argument already?

Reply:
I put the new argument where it is to avoid having too many defaulted bools, as that can get confusing. That said, it does decrease the amount of churn to put it at the end, so I'll go with your suggestion.

Note: we could also do this without the boolean at all by looking for `::` in the type name, but that felt a bit too magical to me.

Reply:
I agree about the magic-ness, that seems like it would be confusing quick.  I too hate lists of bools, and wish we could replace these with enums, but I haven't been able to find the time to hack that sort of thing into the tablegen.

Reply:
> Should we also consider updating https://clang.llvm.org/docs/InternalsManual.html#arguments to explain this particular argument in more detail given that it's a pretty complicated argument already?

Looks like this doesn't go into detail about the types of arguments at all currently. I'll put a note on my todo list to update this separately if you don't mind.

Reply:
> > Should we also consider updating https://clang.llvm.org/docs/InternalsManual.html#arguments to explain this particular argument in more detail given that it's a pretty complicated argument already?
> 
> Looks like this doesn't go into detail about the types of arguments at all currently. I'll put a note on my todo list to update this separately if you don't mind.

Yeah, I never documented the argument types because they used to be rather self-documenting in a lot of ways. We should have started updating these docs a while ago though. Doing it in a follow-up is fine by me (and even if you only do `EnumArgument`; I'm not asking you to sign up to document them all unless you want to).